### ISSUE TITLE
Error with customfields parsing 

### ISSUE THREAD

**Issue by imau** [guilhemmarchand/TA-jira-service-desk-simple-addon#29]
Hi,

We can sucessufilly create a ticket without custom fields, but when we try to use custom fields we get a python error related to the json import.

This is the content of the custom fields structure:

"customfield_10105":{"value": "S4","id":10321},"customfield_10002":{"id":4"}

This is the error we get:

```Bash
2020-08-07 15:24:07,894 ERROR pid=14550 tid=MainThread file=cim_actions.py:message:425 | sendmodaction - worker="xxx-splunk01" signature="json loads failed to accept some of the characters, raw json data before json.loads:={
"fields": {
"project":
 {
"key": "XXX"
 },
"summary": "Splunk Alert: Teste",
"description": "The alert condition for 'JIRA - Open Ticket' was triggered.",
"issuetype": {
"name": "Incident"
},
 "priority" : {
"name": "High"
 },
 \"customfield_10105\":{"value": "S4\",\"id\":10321},\"customfield_10002\":{\"id\":4"}

}
}" action_name="jira_service_desk" search_name="JIRA - Open Ticket" sid="scheduler_am9hby5nYWx2YW
```

The custom fields are added with somewhat random escapes.

Best Regards
Tiago

**Comment by guilhemmarchand** [comment]
Hi @imau 

You custom field structure does not seem to be right to me, here is an example that handles all different type of JIRA custom fields:

`"customfield_10052": {"value": "$result.test_single_choice$"},
"customfield_10048": "$result.test_singleline_text$",
"customfield_10053": [ {"value": "$result.test_multi_choice_grp1$" }, {"value": "$result.test_multi_choice_grp2$" }],
"customfield_10058": $result.test_number$,
"customfield_10049": "$result.test_multiline_text$",
"customfield_10050": "$result.test_url$",
"customfield_10054": "$result.test_date$",
"customfield_10055": "$result.test_datetime$",`

First the following is incorrect:

{"id":4"}

And:

customfield_10105 is suspicious too.

**Comment by guilhemmarchand** [comment]
In:

https://ta-jira-service-desk-simple-addon.readthedocs.io/en/latest/userguide.html#jira-custom-fields

I am referencing some examples and the link to JIRA documentation.
If you can double check the structure, obviously an error in the Add-on code is always possible but I suspect first the structure of your field.

**Comment by imau** [comment]
Hi,

Sorry the first one an error copying here, I did some trying with spaces and things like that and. With the " before the 4 I still get an python error because of the escapes.


So I tested with a just one field in the splunk alert custom fields:

"customfield_10105":{"id":"10321"}

and I get this error, related to json load in python

```
2020-08-07 16:41:07,723 ERROR pid=61174 tid=MainThread file=cim_actions.py:message:425 | sendmodaction - worker="xxx-splunk01" signature="json loads failed to accept some of the characters, raw json data before json.loads:={
"fields": {
"project":
 {
"key": "XXXX"
 },
"summary": "Splunk Alert: Teste",
"description": "The alert condition for 'JIRA - Open Ticket' was triggered.",
"issuetype": {
"name": "Incident"
},
 "priority" : {
"name": "High"
 },
 \"customfield_10105\":{\"id\":\"10321"}

}
}" action_name="jira_service_desk" search_name="JIRA - Open Ticket" sid="scheduler_am9hby5nYWx2YW8__search__RMD5d7d903a21b55139d_at_1596818460_3193" rid="0" app="search" user="xxxxx" digest_mode="0" action_mode="saved" action_status="failure"
```

As you can see the custom field is added to the json with escapes in the " but not in the last one.

Best regards
Tiago

**Comment by guilhemmarchand** [comment]
Right ok 
I will double check shortly.

I think I know where the issue could stand, the Python backend attempts to perform various replacements in the custom field block, the purpose is to protect failure due to any unexpected content, but it is trickier for the custom fields.

One option I am considering would be providing a dropdown to disable this on demand

I will follow up soon

**Comment by guilhemmarchand** [comment]
Hi @imau 

Please upgrade the application to the new release version 1.0.17 available in Splunk Base.

There is a new option on a per alert basis which you can use to disable the custom fields parsing, as explained previsouly this can potentially happen with cases that would not be handled properly.

![image](https://user-images.githubusercontent.com/7503059/89800888-48371b80-db27-11ea-8a31-4ea574503afd.png)

Thank you in advance for your feedback on this fix ;-)

Guilhem

**Comment by guilhemmarchand** [comment]
Jjust noticed the "would be need" error damned... never mind will be fixed in 1.0.18 ;-)

**Comment by imau** [comment]
@guilhemmarchand  

Thank you very much.
